extensions/amp-story/1.0/test/test-amp-story-quiz.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryQuiz} from '../amp-story-quiz';
+
+/**
+ * Populates the quiz with some number of prompts and some number of options
+ *
+ * @param {Window} win
+ * @param {AmpStoryQuiz} quiz
+ * @param {number} numPrompts
+ * @param {number} numOptions
+ */
+const populateQuiz = (win, quizElement, numPrompts = 1, numOptions = 4) => {
+  for (let i = 0; i < numPrompts; i++) {
+    const prompt = win.document.createElement('h1');
+    prompt.textContent = 'prompt';
+    quizElement.appendChild(prompt);
+  }
+
+  const option = win.document.createElement('option');
+  option.textContent = 'option';
+  for (let i = 0; i < numOptions; i++) {
+    quizElement.appendChild(option.cloneNode());
+  }
+};
+
+/**
+ * Populates the quiz with a prompt and three options
+ *
+ * @param {Window} win
+ * @param {AmpStoryQuiz} quiz
+ */
+const populateStandardQuizContent = (win, quizElement) => {
+  populateQuiz(win, quizElement);
+};
+
+describes.realWin(
+  'amp-story-cta-layer',
+  {
+    amp: {
+      runtimeOn: true,
+      extensions: ['amp-story:1.0'],
+    },
+  },
+  env => {
+    let win;
+    let ampStoryQuiz;
+
+    beforeEach(() => {
+      win = env.win;
+      const ampStoryQuizEl = win.document.createElement('amp-story-quiz');
+
+      win.document.body.appendChild(ampStoryQuizEl);
+      ampStoryQuiz = new AmpStoryQuiz(ampStoryQuizEl);
+    });
+
+    it('should take the html and put it in the shadow DOM', async () => {
```

Comment:
This would fail when ran on a browser that has no shadow DOM support.
Instead, you could check that it takes the configuration and renders a template that's within the `this.quizEl_`. Whether `ths.quizEl_` is in the shadow or light DOM depends on `createShadowRootWithStyle()`, which already has its own tests. You can treat it as a blackbox here, and ignore it for your tests.